services/hexathons/src/routes/event.ts

```diff
@@ -0,0 +1,83 @@
+import { asyncHandler, BadRequestError, checkAbility } from "@api/common";
+import express from "express";
+import { FilterQuery } from "mongoose";
+
+import { EventModel, Event } from "../models/event";
+
+export const eventRoutes = express.Router();
+
+eventRoutes.route("/").get(
+  checkAbility("read", "Event"),
+  asyncHandler(async (req, res) => {
+    const filter: FilterQuery<Event> = {};
+
+    if (req.query.hexathon) {
+      filter.hexathon = String(req.query.hexathon);
+    }
+
+    const events = await EventModel.accessibleBy(req.ability).find(filter);
+
+    return res.send(events);
+  })
+);
+
+eventRoutes.route("/").post(
+  checkAbility("create", "Event"),
+  asyncHandler(async (req, res) => {
+    const existingEvent = await EventModel.findOne({
+      hexathon: req.body.hexathon,
+      name: req.body.name,
+      type: req.body.type,
+      startDate: req.body.startDate,
+      endDate: req.body.endDate,
+      location: req.body.location,
+      tags: req.body.tags ? req.body.tags : [],
```

Comment:
This one should be just find by hexathon and name. An event can't have the same name as another for one hexathon.